intelmq/bots/collectors/mail/collector_mail_attach.py

```diff
@@ -28,11 +31,19 @@ class MailAttachCollectorBot(MailCollectorBot):
     mail_user: str = "<user>"
     rate_limit: int = 60
     subject_regex: str = "<subject>"
+    decrypt: bool = False
```

Comment:
As OpenPGP is not the only e-mail encryption method (S/MIME is also widespread), I suggest naming the parameter `decrypt_openpgp` or similar.

```suggestion
    decrypt_openpgp: Optional[bool] = False
```

Reply:
What about `decrypt_gpg` to be in line with other params? (If you mean it, decrypt_openpgp is alright.)

Reply:
The encryption standard is called OpenPGP. gnupg/`gpg` is the program we are using to decrypt it.

So I think `opengpg` is a better suffix, as the important information is the standard's name, not the name of the program that the bot uses internally.

Analogously, S/MIME is the standard, and `gpgsm` is one of the programs for en-/decryption.

Reply:
Seems legit. Should I then use opengpg for other params too? (Which is longer)

Reply:
I'd switch only `_passphrase` as `gpg_home` is related only to the gnupg itself. In my opinion `_openpgp` sounds generally better, but `_gpg` is also acceptable :sweat_smile:

Reply:
So we have now three attributes:

* decrypt_openpgp
* openpgp_passphrase
* gpg_home

Good luck for the users who are not accustomed to PGP nomenclature maze :D . But I admit it's right now.

Reply:
On the first glance, it looks awful, but glancing at it for five minutes, it all makes sense xD
Luckily, they are grouped in the docs.

If the parameters were hierarchical, they looked more sensible:

- decrypt
  - openpgp
    - enabled: yes
    - passphrase: xxx
    - gpg_home: yyy

The more I think of it, `openpgp_passphrase` is also ambiguous, as OpenPGP allows for asymmetric and symmetric encryption, so it could be the passphrase for symmetric decryption.
(If the parameter was also called `gnupgp_passphrase`, that's again ambiguous, as that could also mean a passphrase for unlocking gnupg itself, eg, the keyring or whatever).

So, finally `openpgp_key_passphrase`?
Advantage: If in the future someone adds a parameter to select a specific key, it would be called `openpgp_key_id` or `openpgp_key_file`, etc.

I hope that's my last thought on this topic :sweat_smile:

Reply:
Hey, I might make them hierarchical, that would be the best!

```
"decrypt.openpgp": bool
"decrypt.openpgp.key_passphrase": str
"decrypt.openpgp.gpg_home": Path
```

Isn't it the best?

IMO, the key cannot be chosen, at least by GnuPG. It either is in the keyring, or not. (For encrypting, we choose keys, not for decrypting.)